litellm/proxy/guardrails/guardrail_hooks/acuvity.py

```diff
@@ -0,0 +1,228 @@
+import os
+from typing import List, Literal, Optional, Union
+
+from acuvity import Acuvity, GuardConfig, GuardName, Matches, ResponseMatch, ScanResponseMatch, Security
+from fastapi import HTTPException
+
+from litellm._logging import verbose_proxy_logger
+from litellm.caching.caching import DualCache
+from litellm.integrations.custom_guardrail import (
+    CustomGuardrail,
+    log_guardrail_information,
+)
+from litellm.proxy._types import UserAPIKeyAuth
+from litellm.utils import EmbeddingResponse, ImageResponse, ModelResponse, StreamingChoices
+
+GUARDRAIL_NAME = "acuvity"
+
+class AcuvityGuardrail(CustomGuardrail):
+    def __init__(
+        self, api_key: Optional[str] = None, api_base: Optional[str] = None,
+        **kwargs,
+    ):
+        try:
+            if kwargs.get("event_hook") == "pre_call":
+                self.pre_guard_config_dict = kwargs.pop("vendor_params", None)  # None as default if not found
+                self.pre_guard_config = GuardConfig(self.pre_guard_config_dict)
+            if kwargs.get("event_hook") == "during_call":
+                self.during_guard_config_dict = kwargs.pop("vendor_params", None)  # None as default if not found
+                self.during_guard_config = GuardConfig(self.during_guard_config_dict)
+                if self.during_guard_config.redaction_keys:
+                    raise ValueError("acuvity guard config cannot do redaction for during_call mode, please add redactions in the pre_call")
+            if kwargs.get("event_hook") == "post_call":
+                self.post_guard_config_dict = kwargs.pop("vendor_params", None)  # None as default if not found
+                self.post_guard_config = GuardConfig(self.post_guard_config_dict)
+        except Exception as e:
+            raise ValueError("Acuvity guard config cannot be parsed") from e
+
+        self.acuvity_client = Acuvity(Security(token=(api_key or os.environ["ACUVITY_TOKEN"])))
+        super().__init__(**kwargs)
+
+    @log_guardrail_information
+    async def async_pre_call_hook(
+        self,
+        user_api_key_dict: UserAPIKeyAuth,
+        cache: DualCache,
+        data: dict,
+        call_type: Literal[
+            "completion",
+            "text_completion",
+            "embeddings",
+            "image_generation",
+            "moderation",
+            "audio_transcription",
+            "pass_through_endpoint",
+            "rerank",
+        ],
+    ) -> Optional[Union[Exception, str, dict]]:
+        """
+        Runs before the LLM API call
+        Runs on only Input
+        We just update the data here if there is a pii detection else if more detections are detected then we raise exception.
```

Comment:
I would say something along these lines here:

For configured guards with redactions we mutate outgoing messages. For other triggered guards we raise exceptions.

Reply:
Also, unify the comments on the other hooks in a similar way for a crisp description